packages/backend-core/src/context/utils.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  DEFAULT_TENANT_ID,
+  getAppId,
+  getTenantIDFromAppID,
+  updateTenantId,
+} from "./index"
+import cls from "./FunctionContext"
+import { IdentityContext } from "@budibase/types"
+import { ContextKeys } from "./constants"
+import { dangerousGetDB, closeDB } from "../db"
+import { isEqual } from "lodash"
+import { getDevelopmentAppID, getProdAppID } from "../db/conversions"
+import env from "../environment"
+
+export async function updateUsing(
+  usingKey: string,
+  existing: boolean,
+  internal: (opts: { existing: boolean }) => Promise<any>
+) {
+  const using = cls.getFromContext(usingKey)
+  if (using && existing) {
+    cls.setOnContext(usingKey, using + 1)
+    return internal({ existing: true })
+  } else {
+    return cls.run(async () => {
+      cls.setOnContext(usingKey, 1)
+      return internal({ existing: false })
```

Comment:
this and the internal call in line 23 are async function, should we await for them?

Reply:
returning from an `async` function replaces the async promise return with the returned promise call, same functionality as with the other commented line!

Reply:
Just to be clear, `async () => { return await promise() }` is exactly the same as `async () => { return promise() }` - all the `async` keyword does is enforce the function returns a promise, but if you return a promise specifically from an async function then it is automatically included in the main async function response. Its a quirk of the async definition which can be used for benefit sometimes, like in these situations where I don't want to handle any error scenarios from these functions myself.